GearRent/Networking/APIManager.m

```diff
@@ -0,0 +1,144 @@
+            }
+        } else{
+            [outsideGeohashes addObject:currGH];
+        }
+    }
+    return insideGeohashes;
+}
+
++ (BOOL)polygonContainsGeohash:(NSArray *)coordinates geohash:(GNGeoHash *)geohash {
+    NSMutableArray *currGHCorners = [self getFourConersOfGeohash:geohash];
+    for(int i = 0; i < currGHCorners.count; i ++){
+        CLLocation *currCorner = currGHCorners[i];
+        if(![self polygonContainsPoint:coordinates point:currCorner]){
+            return NO;
+        }
+    }
+    return YES;
+}
+
++ (BOOL)polygonContainsPoint:(NSArray *)coordinates point:(CLLocation *)point {
+    BOOL inside = NO;
+    NSMutableArray *edges = [self getPolygonEdges:coordinates];
+    for(int i = 0; i < edges.count; i ++){
+        Edge *curr = (Edge *)edges[i];
+        CLLocation *lowestPoint = curr.start; // A
+        CLLocation *highestPoint = curr.end; // B
+        if(lowestPoint.coordinate.latitude > highestPoint.coordinate.latitude){
+            lowestPoint = curr.end;
+            highestPoint = curr.start;
+        }
+        if([self isPointSameHeightAsEdge:curr point:point]){
+            point = [[CLLocation alloc] initWithLatitude:point.coordinate.latitude + DBL_EPSILON longitude:point.coordinate.longitude];
+        }
+        if(point.coordinate.latitude <= highestPoint.coordinate.latitude && point.coordinate.latitude >= lowestPoint.coordinate.latitude && point.coordinate.longitude <= MAX(lowestPoint.coordinate.longitude, highestPoint.coordinate.longitude)){
+            if(point.coordinate.longitude >= MIN(lowestPoint.coordinate.longitude, highestPoint.coordinate.longitude)){
+                double slopeOfEdge = 0.0;
+                double slopeOfPoint = 0.0;
+                @try {
+                    slopeOfEdge = (highestPoint.coordinate.latitude - lowestPoint.coordinate.latitude) / (highestPoint.coordinate.longitude - lowestPoint.coordinate.longitude);
+                } @catch (NSException *exception) {
+                    slopeOfEdge = DBL_MAX;
+                }
+                @try{
+                    slopeOfPoint = (point.coordinate.latitude - lowestPoint.coordinate.latitude) / (point.coordinate.longitude - lowestPoint.coordinate.longitude);
+                } @catch (NSException *exception){
+                    slopeOfPoint = DBL_MAX;
+                }
+                if(slopeOfPoint >= slopeOfEdge){ // ray intersects with edge
+                    inside = !inside;
+                }
+            }else{ // ray intersects with edge
+                inside = !inside;
+            }
+        }
+    }
+    return inside;
+}
+
++ (BOOL)isPointSameHeightAsEdge:(Edge *)edge point:(CLLocation *)point {
+    if(fabs(point.coordinate.latitude - edge.start.coordinate.latitude) < DBL_EPSILON ||
+       fabs(point.coordinate.latitude - edge.end.coordinate.latitude) < DBL_EPSILON){
+        return YES;
+    }
+    return NO;
+}
+
++ (NSMutableArray *)getPolygonEdges:(NSArray *)coordinates {
```

Comment:
Can you type these array so that you don't have to force cast? You can do it by changing the signature of this method

```OBJC
(NSMutableArray<Edge *> *)getPolygonEdges:(NSArray<CLLocation *> *)coordinates
```